### PR TITLE
support nested fields in Pulsar presto connector

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Support nested fields in POJOs to be able to be queried by sql

### CODE REVIEW

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/AvroSchemaHandler.java [hunk 1]
```diff
@@ -55,9 +55,23 @@ public Object deserialize(byte[] bytes) {
...
+                    curr = ((GenericRecord) curr).get(positionIndices[i]);
+                    if (curr == null) {
+                        return null;
+                    }
+                }
+            }
+            return curr;
         } catch (Exception ex) {
-            log.error(ex);
+            log.debug(ex,"%s", ex);
```
Comment (sijie) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/AvroSchemaHandler.java]: interesting... why we were using airlift logger not slf4j logger?

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/JSONSchemaHandler.java [hunk 2]
```diff
@@ -81,7 +91,7 @@ public Object extractField(int index, Object currentRecord) {
                 return null;
             }
         } catch (Exception ex) {
-            log.error(ex);
+            log.debug(ex,"%s", ex);
```
Comment (sijie) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/JSONSchemaHandler.java]: same as above

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/AvroSchemaHandler.java (hunk 1 above)
Comment (jerrypeng) [reply pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/AvroSchemaHandler.java]: Just to keep things consistent with the presto project, so when we want to move the connector to the presto repo we can do it with little code modification

Comment (jerrypeng) [reply pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/AvroSchemaHandler.java]: Also I didn't want to introduce more dependencies than necessary and the logger already comes with airlift bootstrap that we need to depend on anyways

**Pull request merged**